### PR TITLE
New tests

### PR DESCRIPTION
**Pull request by zsunberg**
Given how many times we've needed to fix TMazes, I thought we could make some better tests. What do you guys think of these? They test more things like the initial state distribution and the support of all the distributions, etc.

### CODE REVIEW

**Comment by zsunberg** [comment]
I need to create the Project.toml and write more tests to test the different branches but wanted to see if there are any big suggestions first

File: src/model.jl [hunk 1]
```diff
@@ -1,9 +1,137 @@
...
+                        @info "Transition probability negative ($p < 0.0)." s a sp
+                        ok = false
+                    elseif p > 0.0 && !(sp in sup)
+                        @info "State $sp with probability $p is not in support" s a
+                        ok = false
+                    end
+                    psum += p
+                end
+                if !isapprox(psum, 1.0)
+                    @info "Transition probabilities sum to $psum, not 1." s a
```
Comment (rejuvyesh) [inline src/model.jl]: Maybe more like `@warn`?

Comment (rejuvyesh) [reply src/model.jl]: Same at others when `false`.

**Review by rejuvyesh — CHANGES_REQUESTED** [review]
This is great! Thanks for doing this!

**Comment by zsunberg** [comment]
@rejuvyesh @MaximeBouton any other suggestions? If not I will merge and tag this on Monday

**Review by MaximeBouton — APPROVED** [review]

**Comment by MaximeBouton** [comment]
That looks much better thanks @zsunberg !

**Comment by rejuvyesh** [comment]
Looks great and thanks for doing this! :shipit:

**Review by rejuvyesh — APPROVED** [review]

**Pull request merged**